sdk/textanalytics/ai-text-analytics/src/textAnalyticsResult.ts

```diff
@@ -1,13 +1,70 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { TextDocumentStatistics, TextAnalyticsError } from "./generated/models";
+import {
+  TextDocumentStatistics,
+  TextAnalyticsError as GeneratedTextAnalyticsErrorModel,
+  InnerError
+} from "./generated/models";
 
 /**
  * The result of a text analytics operation on a single input document.
  */
 export type TextAnalyticsResult = TextAnalyticsSuccessResult | TextAnalyticsErrorResult;
```

Comment:
In the other languages we have a helper property `isError` or `HasError` . Will a helper like this make sense in JS? 
that way a user can do `if (!errorResult.isError)` instead of `if (errorResult.error === undefined)`

Reply:
In JS we can just do

```js
if (result.error) {
  // result is sure to be a success variant here
}
```

since something that is `undefined` is falsy.

For users using TypeScript, they will even get a strong type bound for `result: TextAnalyticsSuccessResult` in the body of the if, which is why we have the `error?: undefined` in the definition of `TextAnalyticsSuccessResult`. It is "defined to be undefined" for the success type.

Reply:
Cool. Thanks for explaining